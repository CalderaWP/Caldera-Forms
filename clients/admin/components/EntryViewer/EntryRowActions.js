import React from "react";
import PropTypes from 'prop-types';
import {Button} from '@wordpress/components'

export const EntryRowActions = (props) => {

		return(
			<div>
				<Button
					className={EntryRowActions.classNames.view}
					isDefault
					isLarge
					onClick={() => {
						props.onView()
					}}
				>
					View
				</Button>
				<Button
					className={EntryRowActions.classNames.delete}
					isDefault
					isLarge
					onClick={() => {
						props.onDelete()
					}}

				>
					Delete
				</Button>
				<Button
					className={EntryRowActions.classNames.resend}
					isDefault
					isLarge
					onClick={() => {
						props.onResend()
					}}

				>
					Resend
				</Button>
			</div>
		)





};

EntryRowActions.propTypes = {
	onView: PropTypes.func,
	onDelete: PropTypes.func,
	onResend: PropTypes.func,
};

EntryRowActions.classNames = {
	view: 'entry-view-action-view',
	delete: 'entry-view-action-delete',
	resend: 'entry-view-action-resend',
};



