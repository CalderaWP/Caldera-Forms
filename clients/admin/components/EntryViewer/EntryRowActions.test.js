import renderer from "react-test-renderer";
import React from 'react';
import {EntryRowActions} from "./EntryRowActions";
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

describe( 'Entry Row actions component', () => {
	it( 'matches snapshot', () => {
		expect(
			renderer.create(
				<EntryRowActions
					onEntryAction={() => {}}
				/>
			).toJSON()
		).toMatchSnapshot()
	});

	it( 'emits view event', () => {
		let eventReceived = false;
		const component = shallow(
			<EntryRowActions
				onView={(eventType) => {
					eventReceived = true;
				}}
			/>
		);
		component.find( '.' + EntryRowActions.classNames.view).simulate('click');
		expect(eventReceived).toEqual(true);
	});

	it( 'emits delete event', () => {
		let eventTypeReceived = '';
		const component = shallow(
			<EntryRowActions
				onDelete={(eventType) => {
					eventTypeReceived = eventType;
				}}
			/>
		);
		component.find( '.' + EntryRowActions.classNames.delete).simulate('click');
		expect(eventTypeReceived).toEqual('delete');
	});

	it( 'emits view resend', () => {
		let eventTypeReceived = '';
		const component = shallow(
			<EntryRowActions
				onResend={(eventType) => {
					eventTypeReceived = eventType;
				}}
			/>
		);
		component.find( '.' + EntryRowActions.classNames.resend).simulate('click');
		expect(eventTypeReceived).toEqual('resend');
	});
});