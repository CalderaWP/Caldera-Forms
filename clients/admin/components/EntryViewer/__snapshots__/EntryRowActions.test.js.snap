// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Row actions component matches snapshot 1`] = `
<div>
  <button
    autoFocus={undefined}
    className="components-button entry-view-action-view is-button is-default is-large"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    View
  </button>
  <button
    autoFocus={undefined}
    className="components-button entry-view-action-delete is-button is-default is-large"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Delete
  </button>
  <button
    autoFocus={undefined}
    className="components-button entry-view-action-resend is-button is-default is-large"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Resend
  </button>
</div>
`;
